bin/ExtractGName.R

```diff
@@ -7,7 +10,7 @@ zip.fname <- paste(fname, 'zip', sep='.')
 heatmap.dat <- file.path(fname, 'heatmap.RData')
 load(unz(zip.fname, heatmap.dat))
 
-gene.list <- strsplit(go.list[[1]], ':')
-gname.list <- sapply(gene.list, function(x) x[1])
+gene.list <- as.data.frame(go.list, stringsAsFactors = FALSE)
```

Comment:
There are several issues with this line: 

1. When gene ranking algorithms other than k-means are used, go.list[[2]] is an empty list and conversion will result in zero rows for the 2nd column.
1. If multiple gene lists are used and they are of different sizes, this will generate an error. If the gene lists happen to be the same size, they will be "flattened" side-by-side as columns. This may not be a desired outcome.